File: src/problems/leetcode-cn/3634. 使数组平衡的最少移除数目/solution/java/Solution.java
class Solution {
    public int minRemoval(int[] nums, int k) {
        int length = nums.length;
        if (length < 2) return 0;

        Arrays.sort(nums);

        int l = 0, r = 1, inc = 0;
        while (r < length) {
            if ((long) nums[r] <= (long) nums[l]* (long) k) {
                int t = r-l+1;
                inc = t > inc ? t : inc;
                r++;
            } else {
                l++;
            }
        }

        return length-inc;
    }
}

File: src/problems/leetcode-cn/3634. 使数组平衡的最少移除数目/solution/rust/solution.rs
impl Solution {
    pub fn min_removal(mut nums: Vec<i32>, k: i32) -> i32 {
        let length = nums.len();
        if length < 2 {
            return 0;
        }

        nums.sort_unstable();
        let (mut l, mut r, mut inc) = (0, 1, 0);

        while r < length {
            if nums[r] as i64 <= (nums[l] as i64 * k as i64) {
                inc = inc.max(r-l+1);
                r += 1;
            } else {
                l += 1;
            }
        }

        (length-inc) as i32
    }
}